cashctrl_ledger/ledger.py

```diff
@@ -375,17 +318,15 @@ def ledger(self) -> pd.DataFrame:
             pd.DataFrame: A DataFrame with LedgerEngine.ledger() column schema.
         """
         ledger = self._client.list_journal_entries()
-        accounts = self._client.list_accounts()
-        account_map = accounts.set_index('id')['number'].to_dict()
 
         # Individual ledger entries represent a single transaction and
         # map to a single row in the resulting data frame.
         individual = ledger[ledger['type'] != 'COLLECTIVE']
         result = pd.DataFrame({
             'id': individual['id'],
             'date': individual['dateAdded'].dt.date,
-            'account': [account_map[account] for account in individual['creditId']],
-            'counter_account': [account_map[account] for account in individual['debitId']],
+            'account': [self._client.account_from_id(id) for id in individual['creditId']],
+            'counter_account': [self._client.account_from_id(id) for id in individual['debitId']],
             'amount': individual['amount'],
             'currency': individual['currencyCode'],
```

Comment:
Does 'currencyCode' contain the currency in string format ('USD', 'EUR', etc.)?

Is there also a field 'currency' with the currency is text?

Reply:
Yeah, I`ve checked this out, no need to map this field